app/assets/javascripts/Components/course_summary_chart.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { render } from 'react-dom';
+
+import { Bar } from 'react-chartjs-2';
+
+
+export class CourseSummaryChart extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // summary: {
+      //   average: null,
+      //   median: null
+      // },
+      summary: [],
+      data: {
+        labels: [],
+        datasets: [],
+      },
+    };
+  }
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = () => {
+    $.ajax({
+      url: Routes.grade_distribution_course_summaries_path(),
+      type: 'GET',
+      dataType: 'json'
+    }).then(res => {
+      var average;
+      var median;
+      for (const [index, element] of res["datasets"].entries()){
+        element["label"] = I18n.t("main.weighted_total_grades") + " " + res["marking_schemes_id"][index]
+        element["backgroundColor"] = colours[index]
+      }
+      var data = {
+        labels: res['labels'],
+        datasets: res['datasets']
+      }
+      this.setState({data: data})
+      this.setState({summary: res['average']})

```

Comment:
Don't do two consecutive `setState` calls. Instead, create just one combined object (with both `data` and `summary`) and pass that into a single call to `setState`.